diff --git a/plugin/fastfold.vim b/plugin/fastfold.vim
index fd76d25..3ddbebd 100644
--- a/plugin/fastfold.vim
+++ b/plugin/fastfold.vim
@@ -175,11 +175,15 @@ function! s:init()
     " Make &l:foldmethod local to Buffer and NOT Window.
     " UpdateBuf/Win(1) = skip if another session is still loading.
     autocmd BufEnter,WinEnter *
-          \ if exists('b:lastfdm') | let w:lastfdm = b:lastfdm | call s:LeaveWin() | call s:EnterWin() | endif
+          \ if !pumvisible() |
+          \   if exists('b:lastfdm') | let w:lastfdm = b:lastfdm | call s:LeaveWin() | call s:EnterWin() | endif |
+          \ endif
     autocmd BufLeave,WinLeave             *
-          \ call s:LeaveWin() | call s:EnterWin() |
-          \ if exists('w:lastfdm')     | let b:lastfdm = w:lastfdm |
-          \ elseif exists('b:lastfdm') | unlet b:lastfdm | endif
+          \ if !pumvisible() |
+          \   call s:LeaveWin() | call s:EnterWin() |
+          \   if exists('w:lastfdm')     | let b:lastfdm = w:lastfdm |
+          \   elseif exists('b:lastfdm') | unlet b:lastfdm | endif |
+          \ endif
     autocmd BufEnter,WinEnter *
           \ if &l:foldmethod isnot# 'diff' && exists('b:predifffdm') | call s:UpdateBuf(0) | endif
     autocmd BufLeave,WinLeave *
