diff --git a/sources/PseudoTerminal.m b/sources/PseudoTerminal.m
index fa2195ef..86be8a06 100644
--- a/sources/PseudoTerminal.m
+++ b/sources/PseudoTerminal.m
@@ -396,9 +396,17 @@ static NSRect iTermRectCenteredVerticallyWithinRect(NSRect frameToCenter, NSRect
         case WINDOW_TYPE_BOTTOM_PARTIAL:
         case WINDOW_TYPE_LEFT_PARTIAL:
         case WINDOW_TYPE_RIGHT_PARTIAL:
-        case WINDOW_TYPE_NO_TITLE_BAR:
             return mask | NSBorderlessWindowMask | NSResizableWindowMask;
 
+        case WINDOW_TYPE_NO_TITLE_BAR:
+            return (mask |
+                    NSFullSizeContentViewWindowMask |
+                    NSTitledWindowMask |
+                    NSClosableWindowMask |
+                    NSMiniaturizableWindowMask |
+                    NSResizableWindowMask |
+                    NSTexturedBackgroundWindowMask);
+
         case WINDOW_TYPE_TRADITIONAL_FULL_SCREEN:
             return mask | NSBorderlessWindowMask;
 
@@ -638,7 +646,7 @@ static NSRect iTermRectCenteredVerticallyWithinRect(NSRect frameToCenter, NSRect
         [myWindow setFrame:initialFrame display:NO];
     }
 
-    [myWindow setHasShadow:(windowType == WINDOW_TYPE_NORMAL)];
+    [myWindow setHasShadow:(windowType == WINDOW_TYPE_NORMAL) || (windowType == WINDOW_TYPE_NO_TITLE_BAR)];
 
     DLog(@"Create window %@", myWindow);
 
@@ -689,6 +697,17 @@ static NSRect iTermRectCenteredVerticallyWithinRect(NSRect frameToCenter, NSRect
     }
 
     [self updateTabBarStyle];
+    // If the window type is borderless rounded...
+    if(windowType == WINDOW_TYPE_NO_TITLE_BAR) {
+        // Hide the titlebar
+        NSWindow *terminalWindow = self.window;
+        terminalWindow.titlebarAppearsTransparent = YES;
+        terminalWindow.titleVisibility = NSWindowTitleHidden;
+        terminalWindow.showsToolbarButton = NO;
+        [[terminalWindow standardWindowButton:NSWindowCloseButton] setHidden:YES];
+        [[terminalWindow standardWindowButton:NSWindowMiniaturizeButton] setHidden:YES];
+        [[terminalWindow standardWindowButton:NSWindowZoomButton] setHidden:YES];
+    }
     self.window.delegate = self;
 
     [[NSNotificationCenter defaultCenter] addObserver:self
