diff --git a/pythonx/UltiSnips/snippet_manager.py b/pythonx/UltiSnips/snippet_manager.py
index 3d2a2a5..61fc27f 100644
--- a/pythonx/UltiSnips/snippet_manager.py
+++ b/pythonx/UltiSnips/snippet_manager.py
@@ -366,7 +366,7 @@ class SnippetManager(object):
         _vim.command(
             'autocmd InsertLeave * call UltiSnips#LeavingInsertMode()')
 
-        _vim.command('autocmd BufLeave * call UltiSnips#LeavingBuffer()')
+        _vim.command('autocmd BufLeave * if !pumvisible() | call UltiSnips#LeavingBuffer() | endif')
         _vim.command(
             'autocmd CmdwinEnter * call UltiSnips#LeavingBuffer()')
         _vim.command(
